incubator/mariadb/maria.libsonnet

```diff
@@ -0,0 +1,321 @@
+                timeoutSeconds: 5,
+              },
+              readinessProbe: {
+                httpGet: {
+                  path: "/metrics",
+                  port: "metrics",
+                },
+                initialDelaySeconds: 5,
+                timeoutSeconds: 1,
+              },
+              resources: defaults.metrics.resources,
+            },
+          ];
+
+        {
+          apiVersion: "extensions/v1beta1",
+          kind: "Deployment",
+          metadata: {
+            name: name,
+            namespace: namespace,
+            labels: labels,
+          },
+          spec: {
+            template: {
+              metadata: {
+                namespace: namespace,
+                labels: labels,
+              },
+              spec: {
+                containers: [
+                  {
+                    name: "mariadb",
+                    image: defaults.image,
+                    imagePullPolicy: defaults.imagePullPolicy,
+                    env:
+                      secure(passwordSecretName) + [
+                        {
+                          name: "MARIADB_USER",
+                          value: mariaConfig.user
+                        },
+                        {
+                          name: "MARIADB_DATABASE",
+                          value:  mariaConfig.db
+                        },
+                      ],
+                    ports: [
+                      {
+                        name: "mysql",
+                        containerPort: 3306,
+                      },
+                    ],
+                    livenessProbe: {
+                      exec: {
+                        command: [
+                          "mysqladmin",
+                          "ping",
+                        ],
+                      },
+                      initialDelaySeconds: 30,
+                      timeoutSeconds: 5,
+                    },
+                    readinessProbe: {
+                      exec: {
+                        command: [
+                          "mysqladmin",
+                          "ping",
+                        ],
+                      },
+                      initialDelaySeconds: 5,
+                      timeoutSeconds: 1,
+                    },
+                    resources: defaults.resources,
+                    volumeMounts: [
+                      {
+                        name: "config",
+                        mountPath: "/bitnami/mariadb/conf/my_custom.cnf",
+                        subPath: "my.cnf",
+                      },
+                      {
+                        name: "data",
+                        mountPath: "/bitnami/mariadb",
+                      },
+                    ],
+                  },
+                ] + metricsContainer,
+                volumes: [
+                  {
+                    name: "config",
+                    configMap: {
+                      name: name,
```

Comment:
This needs to be a parameter that defaults to `name`, I think.

Reply:
depressing checkmark